pkg/operator/admission/tlsutil.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"crypto/tls"
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+)
+
+type tlsKeypairReloader struct {
+	certMutex sync.RWMutex
+	cert      *tls.Certificate
+	certPath  string
+	keyPath   string
+}
+
+func (keyPair *tlsKeypairReloader) maybeReload() error {
```

Comment:
Was there example code somewhere that we need to give credit to? Or was this purely from docs?

Reply:
Can you provide a link to their code? Was anything modified here for our purposes?

Reply:
I've provided a link to their code in the file. Only logger statements were modified.